frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/toolbar/MergeParentIntoCurrentBranchAction.java

```diff
@@ -0,0 +1,35 @@
+package com.virtuslab.gitmachete.frontend.actions.toolbar;
+
+import com.intellij.openapi.actionSystem.AnActionEvent;
+import io.vavr.control.Option;
+import lombok.val;
+import org.checkerframework.checker.guieffect.qual.UIEffect;
+
+import com.virtuslab.gitmachete.backend.api.SyncToParentStatus;
+import com.virtuslab.gitmachete.frontend.actions.base.BaseMergeParentIntoBranchAction;
+
+public class MergeParentIntoCurrentBranchAction extends BaseMergeParentIntoBranchAction {
+  @Override
+  public Option<String> getNameOfBranchUnderAction(AnActionEvent anActionEvent) {
+    return getCurrentBranchNameIfManaged(anActionEvent);
+  }
+
+  @Override
+  @UIEffect
+  protected void onUpdate(AnActionEvent anActionEvent) {
+    super.onUpdate(anActionEvent);
+
+    val presentation = anActionEvent.getPresentation();
+    if (!presentation.isVisible()) {
+      return;
+    }
+
+    val isInSyncToParent = getCurrentBranchNameIfManaged(anActionEvent)
```

Comment:
🤔 wasn't the action supposed to be active for **red** edges?

Reply:
Hmmmm actually.... doesn't it conflict with the logic in `super.onUpdate`? where eligible statuses include only `OutOfSync` 🤔

Reply:
yup - that was bugged